packages/dev/serializers/src/glTF/2.0/glTFExporter.ts

```diff
@@ -1973,7 +1975,7 @@ export class _Exporter {
         });
 
         const [exportNodes, exportMaterials] = this._getExportNodes(nodes);
-        return this._glTFMaterialExporter._convertMaterialsToGLTFAsync(exportMaterials, ImageMimeType.PNG, true).then(() => {
+        return this._glTFMaterialExporter._convertMaterialsToGLTFAsync(exportMaterials, this._exportMimeType, true).then(() => {

```

Comment:
At a minimum, we should probably check if the texture has alpha and then force it to PNG. Otherwise, it will result in an incorrect glTF. Also, I think there is a switch statement that needs the new WebP case: https://github.com/christaggart/Babylon.js/blob/7f8b4d3ddda5c19b57ec85ecf0a92e4d8b607e7a/packages/dev/serializers/src/glTF/2.0/glTFMaterialExporter.ts#L1138-L1145. Looking at this further, is this option supposed to force the texture to export as the request mime type or keep existing mime type if the texture specifies? Maybe have another flag that is for `preserveExistingMimeType`?

Reply:
Agree with @bghgary it should not simply override but preserve alpha when required.

Reply:
My goal for our use case was to force the texture to export as the requested mimetype, which is why I didn't do any detection and to force to PNG. If you're setting the export mime type explicitly, you should probably be aware of the textures you're exporting.

Reply:
The lines I pointed to earlier needs to check if this option is set or it will not override.

Reply:
In this case I wonder if a custom callback could not be used instead ?

The use case sounds highly specific to your GLTF files not having alpha and I would prefer either a conservative approach or a custom callback to create the textures.

@bghgary any thoughts ?

Reply:
I'm not sure what you mean. None of this is specific to glTF. Textures can have alpha or not regardless of glTF.

Reply:
The webp size improvements were pretty impressive. It would almost make more sense to just default them all to WebP if the standard supports WebP? No alpha check required.

Reply:
There is a lot of debate on whether WebP is better than JPEG. Just do a search and there are tons of articles, though some information is outdated. I don't know if we want to get into that here. :-)

If you just want to add webp and not jpeg, I'm okay with that. The issue with not overriding texture's mimeType is still there though.

Reply:
👍🏻  I'm good with that. I mean, it currently overrides the texture's mimetype and converts it to PNG regardless of what it currently is (that's how we noticed this.. we put jpegs textures in the top.. and pngs always came out the other side).

Reply:
Unless I'm reading the code wrong, it depends on how the textures were created. If you load a glTF, it will mark the textures with the mimeType it was loaded as. Maybe you are creating the textures a different way.

Reply:
Yeah, might just be something omitted elsewhere. I'm not introducing the jpeg export format, just exposing setting the format desired when exporting, if we wanted to limit that to PNG/WEBP, probably needs a new enum to store those two options that doesn't include JPEG instead of piggybacking on `ImageMimeType`.

Here's how we're creating the material:
```
var createArtworkTexture = function() {
        const artworkMaterial = new BABYLON.PBRMetallicRoughnessMaterial("Art", scene);
        artworkMaterial.baseColor = BABYLON.Color3.White();
        artworkMaterial.metallic = 0;
        artworkMaterial.roughness = 1;
        artworkMaterial.baseTexture = new BABYLON.Texture('https://example.com/sometexture.jpg', scene);
        return artworkMaterial;
    }
```    
Line https://github.com/BabylonJS/Babylon.js/pull/13062/files#diff-5a052c97205700de927a116308a464f477571e758a2851b11ae2364af587bbbaL1976 is what was forcing everything to PNG. Though I admit, being able to use jpeg textures and have them exported to webp is useful.

Reply:
Doing this doesn't set the texture's `mimeType` property. If that property is set, it will export as that mimeType. You could actually use this as a workaround to export as JPEG without any of these changes.

Reply:
Wait, nope, I lied. This isn't available to the PBR simple materials.

Reply:
I think the last comment explains the entire situation the best. If the texture's mime type is set, this mime type is used to export the texture. Otherwise, the passed mime type is used. So while adding WebP as a type is a great addition, it is already possible to set the exported mime type per texture. 
Maybe the naming is confusing - this variable is more of a default export mime type in case no mime type was set.
Just my two cents, I'll fly away now

Reply:
Ah, you're right, I totally missed that in the constructor. I think i can pretty much remove everything except the addition of WebP -- without it, it won't allow specifying webp as a valid export mime type.